git_client.py

```diff
@@ -0,0 +1,63 @@
+import os
+import toml
+import git
+from git.exc import GitCommandError
+import time
+
+def read_repo_url_from_toml(config_file: str) -> str:
+    """Reads the repository URL from a TOML configuration file."""
+    # TODO: properly serialise the config
+    config_data = toml.load(config_file)
+    repo_url = config_data.get("repository", {}).get("url")
+    if repo_url:
+        return repo_url
+    else:
+        raise ValueError("Repository URL not found in the TOML file.")
+
+
+def clone_or_pull_repo(repo_url, clone_dir="repo", check_interval=60):
+    """
+    Continuously tries pulling new changes from the remote repository and returns if there are new
+    changes (i.e. if a pull results in a current HEAD being changed).
+
+    This function should be called in a loop: every time it returns it means there are new changes
+    so we should do something.
```

Comment:
Control flow of "block until there's update" is a bit clunky but I'm somewhat rusty (hehe) with Python, so I'm not aware of any better alternatives